documentation/src/docs/wire-protocol.md

```diff
@@ -11,102 +11,97 @@ You may obtain a copy of the License at
+1.  Segment that was read: String (2 bytes length, followed by that many bytes of Java's Modified UTF-8).
+2.  Offset that was read from: Long (8 bytes).
+3.  Is at Tail: Boolean (1 bit).
+4.  Is at `EndOfSegment`: (1 bit).
+5.  Data: Binary (remaining length in message).
 
-Appending
----------
+The client requests to read from a particular Stream at a particular Offset, it then receives one or more replies in the form of `SegmentRead` messages. These contain the data they requested (assuming it exists). The server decides on more or less data to give the client than it asked for, in as many replies as it sees fit.
 
-### Setup Append - Request
+# Appending
 
-1.  ConnectionId - UUID (16 bytes) Identifies this appender.
-2.  Segment to append to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
+## Setup Append - Request
 
-### Append Setup - Reply
+1.  `ConnectionId`: UUID (16 bytes) Identifies this appender.
+2.  Segment to append: String (2 bytes length, followed by that many bytes of Java's Modified UTF-8).
 
-1.  Segment that can be appended to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  ConnectionId - UUID (16 bytes) Identifies the requesting appender.
-3.  ConnectionOffsetAckLevel - Long (8 bytes) What was the last offset received and stored on this segment for this connectionId (0 if new)
+## Append Setup - Reply
 
-### BeginAppendBlock - Request 
+1.  Segment that can be appended: String (2 bytes length, followed by that many bytes of Java's Modified UTF-8).
+2.  `ConnectionId`: UUID (16 bytes) Identifies the requesting appender.
+3.  `ConnectionOffsetAckLevel`: Long (8 bytes) (The last offset received and stored on this Stream Segment for this `ConnectionId` (0 if new).
 
-Only valid after SetupAppend has already been done successfully.
+## BeginAppendBlock - Request 
 
-1.  ConnectionId - UUID (16 bytes)
-2.  ConnectionOffset - Long (8 bytes) Data written so far over this connection to this segment
-3.  Length of data before EndAppendBlock message - Integer (4 bytes) 
+Only valid after `SetupAppend` has already been done successfully.
 
-### EndAppendBlock- Request
+1.  `ConnectionId`: UUID (16 bytes).
+2.  `ConnectionOffset`: Long (8 bytes) (Data written so far over this connection to this Stream Segment).
+3.  Length of data before `EndAppendBlock` message - Integer (4 bytes). 
 
-1.  ConnectionId - UUID (16 bytes)
-2.  ConnectionOffset - Long (8 bytes) Data written so far over this connection
-3.  Block Length - (4 Bytes) Total size of the block that was written. (Note this may more or less than the number of bytes between the BeginAppendBlock and this message)
+## EndAppendBlock- Request
```

Comment:
This is out of date. It is called AppendBlockEnd and has the following:
        final UUID writerId;
        final int sizeOfWholeEvents;
        final ByteBuf data;
        final int numEvents;
        final long lastEventNumber;

Reply:
Accepted suggestion

Reply:
Accepted suggestion